libqtile/backend/x11/core.py

```diff
@@ -804,19 +802,15 @@ def flush(self):
     def graceful_shutdown(self):
         """Try to close windows gracefully before exiting"""
 
-        def get_interesting_pid(win):
-            # We don't need to kill Internal or Static windows, they're qtile
-            # managed and don't have any state.
-            if not isinstance(win, base.Window):
-                return None
-            try:
-                return win.window.get_net_wm_pid()
-            except Exception:
-                logger.exception("Got an exception in getting the window pid")
-                return None
-
-        pids = map(get_interesting_pid, self.qtile.windows_map.values())
-        pids = list(filter(lambda x: x is not None, pids))
+        try:
+            pids = [
+                pid
+                for win in self.qtile.windows_map.values()
+                if not isinstance(win, base.Internal) and (pid := win.get_pid())
```

Comment:
First walrus operator?

Reply:
I think so! This isn't the prettiest use of it but it means we don't have to do `win.get_pid()` twice. Alternatively we could just do
```py
            pids = []
            for win in self.qtile.windows_map.values():
                if not isinstance(win, base.Internal):
                    if pid := win.get_pid():
                        pids.append(pid)
```
Perhaps easier to read?

Reply:
Yes - that's easier to follow.